src/core/structured-log.js

```diff
@@ -396,6 +396,70 @@
       });
     };
 
+    //
+    // Enable batching for sinks in the pipeline after this function.
+    //
+    self.batch = function (batchOptions) {
+        if (!batchOptions) {
+            batchOptions = {};    
+        }
+        if (!batchOptions.batchSize) {
+            batchOptions.batchSize = 100;
+        }
+        if (!batchOptions.timeDuration) {
+            batchOptions.timeDuration = 1000;
+        }
+
+        var batchedLogEvents = [];
+        var lastFlushTime = (new Date()).getTime();
+
+        return self.pipe(function (evt, next) {
+
+            if (batchFlushTimeout) {
```

Comment:
Moving the declaration of `batchFlushTimeout` above this line would help readability.